.github/copilot-instructions.md

```diff
@@ -0,0 +1,239 @@
+# OpenFoodFacts Server Development Instructions
+
+OpenFoodFacts server (Product Opener) is a Perl web application with Docker containerization, serving the world's largest open food products database. The system uses Make + Docker Compose for builds, npm for frontend tooling, and comprehensive test suites.
+
+Always reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+## ✅ System Status: Fully Operational
+
+**ALL BUILD COMMANDS NOW WORKING**: Docker build and development environment are fully functional after network connectivity issues were resolved through proper URL whitelisting.
+
+**Key Requirements for Success**:
+- Use `DOCKER_BUILDKIT=1` for optimal build performance
+- First-time builds take 15-20 minutes due to extensive Perl dependencies
+- All major development workflows are operational
+
+## Working Effectively
+
+### Initial Setup (Full Development Environment - VERIFIED WORKING)
+1. **Prerequisites**: Ensure Docker and Docker Compose are available
+2. **Clone repository**: `git clone https://github.com/openfoodfacts/openfoodfacts-server.git`
+3. **Frontend dependencies**: `npm install` - takes ~4 seconds (cached) to ~30 seconds (fresh), WORKS
+4. **Frontend build**: `npm run build` - takes ~17 seconds, WORKS  
+5. **Frontend linting**: `npm run lint` - takes ~4 seconds, WORKS
+6. **Full development environment**: `DOCKER_BUILDKIT=1 make dev` - takes ~15-20 minutes first time, WORKS
+7. **Unit tests**: `make unit_test` - runs all backend unit tests, WORKS
+
+### Build and Development Commands Status
+
+**✅ WORKING COMMANDS (FULLY VALIDATED):**
+- `npm install` - Install frontend dependencies (~4 seconds cached, ~30 seconds fresh)
+- `npm run build` - Build frontend assets with Gulp (~17 seconds)  
+- `npm run build:watch` - Auto-rebuild on file changes (use Ctrl+C to stop)
+- `npm run lint` - Lint JavaScript, CSS, and SCSS (~4 seconds)
+- `npm run lint:js` - Lint JavaScript files only
+- `npm run lint:css` - Lint CSS files only  
+- `npm run lint:scss` - Lint SCSS files only
+- `npm run test` - Alias for npm run lint (frontend validation)
```

Comment:
You don't need any of these, we always run npm inside the container

Reply:
@alexgarel